numpy/core/tests/test_getlimits.py

```diff
@@ -112,6 +114,28 @@ def test_known_types():
         assert_ma_equal(ld_ma, _float_ma[128])
 
 
+def test_subnormal_warning():
+    """Test that the subnormal is zero warning is not being raised."""
+    with np.errstate(all='ignore'):
```

Comment:
We need to move this `np.errstate` into the code?  Users shouldn't see those warnings either.

Reply:
I used it as the other test is doing, https://github.com/numpy/numpy/blob/1769352927f60b86bf183706c887df5a892c6f8f/numpy/core/tests/test_getlimits.py#L104.I'm not completely sure why that was necessary before and I can try removing both

Reply:
Oh, sorry.  I didn't realize this was a unit test for `MachAr` and not for the (normal) `MachArLike` branch.  The warning seems indeed OK here, this code shouldn't really be used anyway.  (The only place I know it may get used is when running in valgrind, due to broken longdouble support).